File: TheListCompany/app/src/main/java/com/example/thelistcompany/MainActivity.java
package com.example.thelistcompany;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    private LaunchDataService dataService;
    private Button getDataButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Objects.requireNonNull(getSupportActionBar()).hide();
        setContentView(R.layout.activity_main);
        dataService = new LaunchDataService(MainActivity.this);
        dataService.sendRequest();
        getDataButton = (Button) findViewById(R.id.getButton);
    }

    public void retrieveData(View view) {
        dataService.getLaunchData();
        System.out.println(LaunchDataService.getLaunches());
        navigateToListActivity();
    }

    public void navigateToListActivity() {
        Intent intent = new Intent(MainActivity.this, ListActivity.class);
        startActivity(intent);
    }
}

File: TheListCompany/app/src/main/java/com/example/thelistcompany/MainActivity.kt
package com.example.thelistcompany

import androidx.appcompat.app.AppCompatActivity
import com.example.thelistcompany.LaunchDataService
import android.os.Bundle
import com.example.thelistcompany.R
import android.content.Intent
import android.view.View
import android.widget.Button
import java.util.*

class MainActivity : AppCompatActivity() {
    private var dataService: LaunchDataService? = null
    private var getDataButton: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.hide()
        setContentView(R.layout.activity_main)
        dataService = LaunchDataService(this@MainActivity)
        dataService!!.sendRequest()
        getDataButton = findViewById<View>(R.id.getButton) as Button
    }

    fun retrieveData(view: View?) {
        dataService!!.launchData
        println(LaunchDataService.launches)
        navigateToListActivity()
    }

    private fun navigateToListActivity() {
        val intent = Intent(this@MainActivity, ListActivity::class.java)
        startActivity(intent)
    }
}